shared/scss/abstracts/_variables.scss

```diff
@@ -6,3 +6,73 @@ $fontWeightRegular: 400;
 $fontWeightMedium: 500;
 $fontWeightSemibold: 600;
 $fontWeightBold: 700;
+
+$fontSizeXs: 0.75em;
+$fontSizeSm: 0.875em;
+$fontSizeBase: 1em;
+$fontSizeLg: 1.125em;
+$fontSizeXl: 1.25em;
+$fontSize2Xl: 1.5em;
+$fontSize3Xl: 1.875em;
+$fontSize4Xl: 2.25em;
+// To be continued...
+
+/*------------------------------------*\
+  #BREAKPOINTS
+\*------------------------------------*/
+
+$led4k: 160em;
+$alpha: 87.5em;
+$preAlpha: 75.0625em;
+$beta: 75em;
+$preBeta: 64em;
+$gamma: 62em;
+$lota: 31.25em;
+$preZeta: 47.9375em;
+$eta: 40em;
+$kappa: 25em;
+
+/*------------------------------------*\
+  #SPACES (padding & margin & gap) 
+\*------------------------------------*/
+$space1: calc(0.25em * 1);
+$space2: calc(0.25em * 2);
+$space3: calc(0.25em * 3);
+$space4: calc(0.25em * 4);
+$space5: calc(0.25em * 5);
+$space6: calc(0.25em * 6);
+$space7: calc(0.25em * 7);
+$space8: calc(0.25em * 8);
+$space9: calc(0.25em * 9);
+$space10: calc(0.25em * 10);
+$space11: calc(0.25em * 11);
+$space12: calc(0.25em * 12);
+// To be continued...
```

Comment:
maybe here we can use for loop ?! and you can define for 16 space?!
less code and better code ;)

Reply:
Sass does not support generating variables but we can use lists to do that a for loop to generate the values to the list and we can use loop here because these spaces variables follow a consistant pattern I mean like 1, 2, 3, .... to 16

Reply:
```
@for $i from 1 through 12 {
  $space#{$i}: calc(0.25em * #{$i});
}

```

Reply:
> ```
> @for $i from 1 through 12 {
>   $space#{$i}: calc(0.25em * #{$i});
> }
> ```

it throws this error  : 

```
Error: expected ":".
   ╷
40 │   $space#{$i}: calc(0.25em * #{$i});
   │         ^
   ╵
```